src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -575,5 +575,47 @@ private static IEnumerable<PropertyInfo> GetAllProperties(Type type)
 
             return allProperties;
         }
+
+        private static object GetPropertyValue(PropertyInfo property, object instance, IConfiguration config, BinderOptions options)
+        {
+            string propertyName = GetPropertyName(property);
+            return BindInstance(
+                property.PropertyType,
+                property.GetValue(instance),
+                config.GetSection(propertyName),
+                options);
+        }
+
+        private static string GetPropertyName(MemberInfo property)
+        {
+            if (property == null)
+            {
+                throw new ArgumentNullException(nameof(property));
```

Comment:
This is a breaking change right? As if before we didn't find the property we just didn't bind?

Reply:
@safern shouldn't be. Tests will tell. Before the property was accessed directly so it'd have thrown an null reference exception  if the property wasn't provided. This is just proactive about throwing an exception.